earth2studio/data/arco.py

```diff
@@ -256,6 +252,10 @@ async def fetch(
         await tqdm.gather(
             *func_map, desc="Fetching ARCO data", disable=(not self._verbose)
         )
+
+        # Delete cache if needed
+        if not self._cache:
+            shutil.rmtree(self.cache)
```

Comment:
Just curious, why move this into `fetch`?

Reply:
Reverted and also updated others so they are in call function